File: scripts/prepare_atcoder.py
import shutil
from pathlib import Path
from string import ascii_lowercase

import pyperclip

print("input contest name.")
print("example: abc284")
print()

contest_name = input("type here: ")
contest_dir_path = Path() / "atcoder.jp" / "contests" / contest_name / "tasks"
templates_py_path = Path() / "templates" / "main.py"

for ch in ascii_lowercase[:5]:
    problem_dir_name = contest_name + "_" + ch
    problem_dir_path = contest_dir_path / problem_dir_name
    problem_dir_path.mkdir(parents=True, exist_ok=True)
    shutil.copy(templates_py_path, problem_dir_path)

problem_a = contest_name + "_a"
to_copied = ["acac", "https://" + str(contest_dir_path / problem_a), "-m"]
pyperclip.copy(" ".join(to_copied))  # type: ignore

File: src/bin/prepare_atcoder.rs
use std::{
    fs,
    io::{self, Write},
    path::Path,
};

fn main() {
    println!("input contest name.");
    println!("example: abc284");
    println!();

    let contest_name = input("type here: ");
    let contest_dir_path = Path::new(".")
        .join("atcoder.jp")
        .join("contests")
        .join(&contest_name)
        .join("tasks");
    let templates_py_path = Path::new(".").join("templates").join("main.py");

    for ch in 'a'..='e' {
        let problem_dir_name = contest_name.to_string() + "_" + &ch.to_string();
        let problem_dir_path = contest_dir_path.join(problem_dir_name);
        problem_dir_path.mkdir_all();
        fs::copy(
            &templates_py_path,
            problem_dir_path.join(templates_py_path.file_name().unwrap()),
        )
        .unwrap();
    }
}

fn input(prompt: &str) -> String {
    print!("{}", prompt);
    io::stdout().flush().unwrap();
    let mut buf = String::new();
    io::stdin().read_line(&mut buf).unwrap();
    buf.trim().to_string()
}

trait MyPath {
    fn mkdir_all(&self);
}

impl MyPath for Path {
    fn mkdir_all(&self) {
        fs::create_dir_all(self).unwrap()
    }
}
